### PR TITLE
Add ts/ds test for tbl model

### PR DESCRIPTION
**Pull request by wyxxxcat**

### CODE REVIEW

File: regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    def tableName = "test_ds_tbl_aggregate_table"
+    def test_num = 0
+    def insert_num = 5
+
+    def exist = { res -> Boolean
+        return res.size() != 0
+    }
+
+    sql "DROP TABLE IF EXISTS ${dbName}.${tableName}"
+    sql "DROP TABLE IF EXISTS TEST_${dbName}.${tableName}"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy]: target_sql?

Comment (wyxxxcat) [reply regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy]: sorry, done

File: regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    helper.ccrJobDelete()
+    helper.ccrJobCreate()
+
+    assertTrue(helper.checkRestoreFinishTimesOf("${tableName}", 30))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "sql"))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "target"))
+
+    sql_res = sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy]:
```suggestion
    def sql_res = sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy]:
```suggestion
    def target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/db_sync/table/duplicate/test_ds_tbl_duplicate.groovy [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    helper.ccrJobDelete()
+    helper.ccrJobCreate()
+
+    assertTrue(helper.checkRestoreFinishTimesOf("${tableName}", 30))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "sql"))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "target"))
+
+    sql_res = sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/duplicate/test_ds_tbl_duplicate.groovy]:
```suggestion
    def sql_res = sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/db_sync/table/duplicate/test_ds_tbl_duplicate.groovy [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/duplicate/test_ds_tbl_duplicate.groovy]:
```suggestion
    def target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_ds_tbl_aggregate") {
+    def helper = new GroovyShell(new Binding(['suite': delegate]))
+            .evaluate(new File("${context.config.suitePath}/../common", "helper.groovy"))
+
+    def dbName = context.dbName
+    def tableName = "test_ds_tbl_aggregate_table"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/aggregate/test_ds_tbl_aggregate.groovy]:
```suggestion
    def tableName = "tbl_" + helper.randomSuffix()
```

File: regression-test/suites/db_sync/table/duplicate/test_ds_tbl_duplicate.groovy [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_ds_tbl_duplicate") {
+    def helper = new GroovyShell(new Binding(['suite': delegate]))
+            .evaluate(new File("${context.config.suitePath}/../common", "helper.groovy"))
+
+    def dbName = context.dbName
+    def tableName = "test_ds_tbl_duplicate_table"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/duplicate/test_ds_tbl_duplicate.groovy]:
```suggestion
    def tableName = "tbl_" + helper.randomSuffix()
```

File: regression-test/suites/db_sync/table/unique/test_ds_tbl_unique.groovy [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_ds_tbl_unique") {
+    def helper = new GroovyShell(new Binding(['suite': delegate]))
+            .evaluate(new File("${context.config.suitePath}/../common", "helper.groovy"))
+
+    def dbName = context.dbName
+    def tableName = "test_ds_tbl_unique_table"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/unique/test_ds_tbl_unique.groovy]:
```suggestion
    def tableName = "tbl_" + helper.randomSuffix()
```

File: regression-test/suites/db_sync/table/unique/test_ds_tbl_unique.groovy [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    helper.ccrJobDelete()
+    helper.ccrJobCreate()
+
+    assertTrue(helper.checkRestoreFinishTimesOf("${tableName}", 30))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "sql"))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "target"))
+
+    sql_res = sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/unique/test_ds_tbl_unique.groovy]:
```suggestion
    def sql_res = sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/db_sync/table/unique/test_ds_tbl_unique.groovy [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+    target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/db_sync/table/unique/test_ds_tbl_unique.groovy]:
```suggestion
    def target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/table_sync/table/aggregate/test_ts_tbl_aggregate.groovy [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_ts_tbl_aggregate") {
+    def helper = new GroovyShell(new Binding(['suite': delegate]))
+            .evaluate(new File("${context.config.suitePath}/../common", "helper.groovy"))
+
+    def dbName = context.dbName
+    def tableName = "test_ts_tbl_aggregate_table"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/aggregate/test_ts_tbl_aggregate.groovy]:
```suggestion
    def tableName = "tbl_" + helper.randomSuffix()
```

File: regression-test/suites/table_sync/table/aggregate/test_ts_tbl_aggregate.groovy [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    helper.ccrJobDelete(tableName)
+    helper.ccrJobCreate(tableName)
+
+    assertTrue(helper.checkRestoreFinishTimesOf("${tableName}", 30))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "sql"))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "target"))
+
+    sql_res = sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/aggregate/test_ts_tbl_aggregate.groovy]:
```suggestion
    def sql_res = sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/table_sync/table/aggregate/test_ts_tbl_aggregate.groovy [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+    target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/aggregate/test_ts_tbl_aggregate.groovy]:
```suggestion
    def target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/table_sync/table/duplicate/test_ts_tbl_duplicate.groovy [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_ts_tbl_duplicate") {
+    def helper = new GroovyShell(new Binding(['suite': delegate]))
+            .evaluate(new File("${context.config.suitePath}/../common", "helper.groovy"))
+
+    def dbName = context.dbName
+    def tableName = "test_ts_tbl_duplicate_table"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/duplicate/test_ts_tbl_duplicate.groovy]:
```suggestion
    def tableName = "tbl_" + helper.randomSuffix()
```

File: regression-test/suites/table_sync/table/duplicate/test_ts_tbl_duplicate.groovy [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+    helper.ccrJobDelete(tableName)
+    helper.ccrJobCreate(tableName)
+
+    assertTrue(helper.checkRestoreFinishTimesOf("${tableName}", 30))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "sql"))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "target"))
+
+    sql_res = sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/duplicate/test_ts_tbl_duplicate.groovy]:
```suggestion
    def sql_res = sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/table_sync/table/duplicate/test_ts_tbl_duplicate.groovy [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+
+    target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/duplicate/test_ts_tbl_duplicate.groovy]:
```suggestion
    def target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/table_sync/table/unique/test_ts_tbl_unique.groovy [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite("test_ts_tbl_unique") {
+    def helper = new GroovyShell(new Binding(['suite': delegate]))
+            .evaluate(new File("${context.config.suitePath}/../common", "helper.groovy"))
+
+    def dbName = context.dbName
+    def tableName = "test_ts_tbl_unique_table"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/unique/test_ts_tbl_unique.groovy]:
```suggestion
    def tableName = "tbl_" + helper.randomSuffix()
```

File: regression-test/suites/table_sync/table/unique/test_ts_tbl_unique.groovy [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+    helper.ccrJobDelete(tableName)
+    helper.ccrJobCreate(tableName)
+
+    assertTrue(helper.checkRestoreFinishTimesOf("${tableName}", 30))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "sql"))
+
+    assertTrue(helper.checkShowTimesOf("SHOW TABLES LIKE \"${tableName}\"", exist, 60, "target"))
+
+    sql_res = sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/unique/test_ts_tbl_unique.groovy]:
```suggestion
    def sql_res = sql "SHOW CREATE TABLE ${tableName}"
```

File: regression-test/suites/table_sync/table/unique/test_ts_tbl_unique.groovy [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+
+    target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```
Comment (w41ter) [inline regression-test/suites/table_sync/table/unique/test_ts_tbl_unique.groovy]:
```suggestion
    def target_res = target_sql "SHOW CREATE TABLE ${tableName}"
```

**Review by w41ter — CHANGES_REQUESTED** [review]

**Review by w41ter — APPROVED** [review]

**Review by w41ter — APPROVED** [review]

**Pull request merged**